### PR TITLE
Features/5522 new api for source

### PR DESCRIPTION
**Pull request by utas-raymondng**

### CODE REVIEW

File: geonetwork-api/src/test/java/au/org/aodn/geonetwork_api/openapi/api/helper/SiteHelperTest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     * Function test only, we do not have docker image to run for this test, the
+     * integration test in geonetwork should test it more.
+     */
+    @Test
+    public void verifyGetAllSettingsDetails() {
+        Setting s1 = new Setting();
+        s1.setName(SiteHelper.HOST);
+        s1.setValue("geonetwork-edge.edge.aodn.org.au");
```
Comment (vietnguyengit) [inline geonetwork-api/src/test/java/au/org/aodn/geonetwork_api/openapi/api/helper/SiteHelperTest.java:28]: can the value be managed in the test's properties yaml?

and you can create a variable for being reused at line 49.

File: geonetwork/src/main/java/au/org/aodn/geonetwork4/controller/Api.java [hunk 2]
```diff
@@ -31,21 +41,120 @@
...
+                else if(harvester instanceof OgcWxSHarvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((OgcWxSHarvester) harvester).getParams().url, "/"));
+                    info.put("harvesterType", "OgcWxSHarvester");
+                }
+                else if(harvester instanceof Geonet20Harvester) {
+                    // Will have remote link to logo
+                    info.put("harvesterUri", StringUtils.removeEnd(((Geonet20Harvester) harvester).getParams().host, "/"));
+                    info.put("harvesterType", "Geonet20Harvester");
+                }
```
Comment (vietnguyengit) [inline geonetwork/src/main/java/au/org/aodn/geonetwork4/controller/Api.java:149]: what will happen if the `harvester` is null or not instanceof any declared type? 

and if we can reorganise these to use some suggestions like in this thread: https://stackoverflow.com/questions/5579309/is-it-possible-to-use-the-instanceof-operator-in-a-switch-statement

**Review by vietnguyengit — DISMISSED** [review]
LGTM but I have comments, leave it's up to you to make the changes.

File: geonetwork/src/main/java/au/org/aodn/geonetwork4/controller/Api.java (hunk 2 above)
Comment (utas-raymondng) [reply geonetwork/src/main/java/au/org/aodn/geonetwork4/controller/Api.java]: changed with null check and logging

File: geonetwork-api/src/test/java/au/org/aodn/geonetwork_api/openapi/api/helper/SiteHelperTest.java (hunk 1 above)
Comment (utas-raymondng) [reply geonetwork-api/src/test/java/au/org/aodn/geonetwork_api/openapi/api/helper/SiteHelperTest.java]: Introduce constant

**Review by vietnguyengit — APPROVED** [review]

**Pull request merged**